build-parent/pom.xml

```diff
@@ -681,6 +681,48 @@
                     <groupId>com.diffplug.spotless</groupId>
                     <artifactId>spotless-maven-plugin</artifactId>
                     <version>2.44.4</version>
+                    <configuration>
+                        <java>
+                            <!-- <removeUnusedImports /> -->
+                            <formatAnnotations/>
+                            <eclipse>
+                                <file>
+                                    ${maven.multiModuleProjectDirectory}/independent-projects/ide-config/src/main/resources/eclipse-format.xml
+                                </file>
+                            </eclipse>
+                            <!-- not aligned on apply: alternates order /> -->
+                            <importOrder>
+                                <file>
+                                    ${maven.multiModuleProjectDirectory}/independent-projects/ide-config/src/main/resources/eclipse.importorder
+                                </file>
+                            </importOrder>
+                        </java>
+                        <formats>
+                            <format>
+                                <includes>
+                                    <include>.gitignore</include>
+                                </includes>
+                                <trimTrailingWhitespace/>
```

Comment:
its like 3 plugins in 1:

- https://github.com/checkstyle/checkstyle/pull/16574